pkg/nmi/server/endpointpolicy.go

```diff
@@ -39,13 +43,19 @@ func DeleteEndpointRoutePolicy(podIP string, metadataIP string) error {
 	}
 
 	endpoint, err := getEndpointByIP(podIP)
+
 	if err != nil {
-		return fmt.Errorf("No endpoint found for Pod IP - %s. Error: %w", podIP, err)
+		return fmt.Errorf("Calling HCN agent falied for Pod IP - %s. Error: %v", podIP, err)
+	}
+
+	if endpoint == nil {
+		klog.Warningf("No deleting action: no endpoint found for Pod IP - %s.", podIP)
```

Comment:
>Warningf [](start = 7, length = 8)

Should this be a warning, its still a happy path. There is nothing wrong with this, right?

Reply:
updated